crypto/src/tests/secp256k1_tests.rs

```diff
@@ -145,3 +145,29 @@ async fn signature_service() {
     // Verify the signature we received.
     assert!(pk.verify(digest.as_ref(), &signature).is_ok());
 }
+
+use proptest::arbitrary::Arbitrary;
+
+proptest::proptest! {
+    #[test]
+    fn test_k256_against_secp256k1_lib(
+        r in <[u8; 32]>::arbitrary()
+    ) {
+        let message: &[u8] = b"Hello, world!";
+
+        // contruct private key with bytes and signs message
+        let priv_key = <Secp256k1PrivateKey as ToFromBytes>::from_bytes(&r).unwrap();
+        let key_pair = Secp256k1KeyPair::from(priv_key);
+        let signature = key_pair.sign(&message);
+        assert!(key_pair.public().verify(&message, &signature).is_ok());
+
+        // use k256 to construct private key with the same bytes and signs the same message
+        let priv_key_1 = k256::ecdsa::SigningKey::from_bytes(&r).map_err(|_e| signature::Error::new())?;
+        let pub_key_1 = priv_key_1.verifying_key();
+        let signature_1: k256::ecdsa::Signature = priv_key_1.sign(message);
+        assert!(pub_key_1.verify(message, &signature_1).is_ok());
+
```

Comment:
We might be able to check for the equality of the byte representation of the public keys here as well.